### PR TITLE
Feat: Adds `Button` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- Move Button Component from @faststore/ui to @faststore/components
- Move Button styles from nextjs.store to @faststore/ui.
- Updates Button documentation page on docusaurus

## How to test it?

run `yarn start` in app/docs folder and checks the Button component or test it using the doc [preview link.](https://faststore-git-feat-button-component-fs-662-faststore.vercel.app/reference/ui/atoms/Button)

### CODE REVIEW

File: apps/docs/docs/reference/ui/atoms/Button.mdx [hunk 1]
```diff
@@ -4,56 +4,542 @@ keywords:
...
+    token="--fs-button-transition-timing"
+    value="var(--fs-transition-timing)"
+  />
+</TokenTable>
+
+---
+
+## Hierarchy
+
+### Primay
```
Comment (eduardoformiga) [inline apps/docs/docs/reference/ui/atoms/Button.mdx:102]:
```suggestion
### Primary
```

File: apps/docs/docs/reference/ui/atoms/Button.mdx [hunk 2]
```diff
@@ -4,56 +4,542 @@ keywords:
...
-    <img
-      alt="A vintage camera"
-      src="https://storecomponents.vtex.app/assets/fit-in/480x480/center/middle/https%3A%2F%2Fstorecomponents.vtexassets.com%2Farquivos%2Fids%2F155481%2FFrame-3.jpg%3Fv%3D636793814536230000"
-    />
-    <div>
-    <Button class="bg-green-500">ADD TO CART</Button>
-
-    </div>
-</Card>
+<Button variant="primay">Button</Button>
```
Comment (eduardoformiga) [inline apps/docs/docs/reference/ui/atoms/Button.mdx:105]:
```suggestion
<Button variant="primary">Button</Button>
```

File: apps/docs/src/theme/ReactLiveScope/index.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */
-import '@faststore/styles'
+// import '../../../../../packages/styles'
```
Comment (eduardoformiga) [inline apps/docs/src/theme/ReactLiveScope/index.tsx:7]: hmm, do we need to change this file?

**Review by eduardoformiga — APPROVED** [review]
I tested it [here](https://faststore-aenjf3pis-faststore.vercel.app/reference/ui/atoms/Button) and looks good, Fan! 🎉 


Just a little typo that affects the primary variant. I'll let pre-approved.

File: apps/docs/src/theme/ReactLiveScope/index.tsx (hunk 3 above)
Comment (hellofanny) [reply apps/docs/src/theme/ReactLiveScope/index.tsx:7]: in the future we should remove this import, but let's keep it, it was my bad I was testing and forgot to adjust it :)

**Pull request merged**